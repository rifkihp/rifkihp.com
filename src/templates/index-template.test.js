import React from 'react';
import renderer from 'react-test-renderer';

import IndexTemplate from './index-template';

describe('IndexTemplate', () => {
  const props = {
    data: {
      allMarkdownRemark: {
        edges: [
          {
            node: {
              fields: {
                slug: '/test_0',
                categorySlug: '/test_0',
              },
              frontmatter: {
                date: '2016-09-01',
                description: 'test_0',
                category: 'test_0',
                title: 'test_0',
              },
            },
          },
          {
            node: {
              fields: {
                slug: '/test_1',
                categorySlug: '/test_1',
              },
              frontmatter: {
                date: '2016-09-01',
                description: 'test_1',
                category: 'test_1',
                title: 'test_1',
              },
            },
          },
        ],
      },
      site: {
        siteMetadata: {
          title: 'test',
          subtitle: 'test',
          copyright: 'Copyright',
          author: {
            contacts: {
              email: '#',
              github: '#',
              twitter: '#',
            },
          },
        },
      },
    },
    pageContext: {
      currentPage: 1,
      hasNextPage: true,
      hasPrevPage: true,
      numPages: 1,
      postSlugs: ['/blog/year-one-recap/'],
      sortByNew: true,
    },
  };

  it('renders correctly', () => {
    const tree = renderer.create(<IndexTemplate {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
