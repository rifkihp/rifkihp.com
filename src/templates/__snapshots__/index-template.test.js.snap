// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IndexTemplate renders correctly 1`] = `
Array [
  <StaticQuery
    render={[Function]}
  />,
  <StaticQuery
    render={[Function]}
  />,
]
`;
